moveit_simple/src/moveit_simple.cpp

```diff
@@ -478,6 +595,23 @@ Eigen::Affine3d Robot::transformToBase(const Eigen::Affine3d &in,
 
 
 
+bool Robot::getFK(const std::vector<double> & joint_point,
+                  Eigen::Affine3d &pose) const
+{
+  robot_state_->setJointGroupPositions(joint_group_, joint_point);
+  const std::vector<std::string> link_names = robot_model_ptr_->getLinkModelNames();
+  if (!link_names.empty())
```

Comment:
The number of free/variable joints should capture this: http://docs.ros.org/indigo/api/moveit_core/html/classmoveit_1_1core_1_1RobotState.html#a711eb6a22c6cc8fbe6de0df2290c5bf0

Any fixed joints should not be included in this count